cognite/client/data_classes/time_series.py

```diff
@@ -150,6 +203,57 @@ def asset(self) -> Asset:
         return cast("Asset", self._cognite_client.assets.retrieve(id=self.asset_id))
 
 
+class TimeSeriesWrite(TimeSeriesCore):
+    """This represents a sequence of data points. The TimeSeries object is the metadata about
+    the datapoints, and the Datapoint object is the actual data points. This is the writing version
+    of TimesSeries, which is used when writing to CDF.
+
+    Args:
+        external_id (str): The externally supplied ID for the time series.
+        name (str | None): The display short name of the time series. Note: Value of this field can differ from name presented by older versions of API 0.3-0.6.
+        is_string (bool | None): Whether the time series is string valued or not.
+        metadata (dict[str, str] | None): Custom, application-specific metadata. String key -> String value. Limits: Maximum length of key is 32 bytes, value 512 bytes, up to 16 key-value pairs.
+        unit (str | None): The physical unit of the time series.
+        unit_external_id (str | None): The physical unit of the time series (reference to unit catalog). Only available for numeric time series.
+        asset_id (int | None): Asset ID of equipment linked to this time series.
+        is_step (bool | None): Whether the time series is a step series or not.
+        description (str | None): Description of the time series.
+        security_categories (Sequence[int] | None): The required security categories to access this time series.
+        data_set_id (int | None): The dataSet ID for the item.
+        legacy_name (str | None): Set a value for legacyName to allow applications using API v0.3, v04, v05, and v0.6 to access this time series. The legacy name is the human-readable name for the time series and is mapped to the name field used in API versions 0.3-0.6. The legacyName field value must be unique, and setting this value to an already existing value will return an error. We recommend that you set this field to the same value as externalId.
+    """
+
+    def __init__(
+        self,
+        external_id: str,
```

Comment:
ExternalID is not required by the API, but I think this is because of legacy reasons. So suggest we require it in the SDK.